packages/fuels/tests/logs.rs

```diff
@@ -1259,3 +1258,37 @@ async fn contract_token_ops_error_messages() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_log_results() -> Result<()> {
```

Comment:
You can make a wrong log by using assembly and the `log` instruction and putting a invalid log id  in `rb`. This will give you the same result without the two additional projects.

Reply:
AFAIK the log decoder is bound to a contract response. So he would need to make a dummy contract call just to get access to a log decoder which he can use on the receipts of his custom script with `log `. Is there a more straight up way to do this? Otherwise I would keep it as it is

Reply:
Maybe we should leave this as it is for now until Ahmed moves the log decoder?

Reply:
> On a related note, I'm thinking of moving the log decoder to the contract instance so that user can decode logs just from receipts that they retrieved via a tx id.

@MujkicA This is an excellent idea! Are you on it already, and/or do we already have an issue documenting this?

Reply:
> AFAIK the log decoder is bound to a contract response. So he would need to make a dummy contract call just to get access to a log decoder which he can use on the receipts of his custom script with `log `. Is there a more straight up way to do this? Otherwise I would keep it as it is.
> 
> On a related note, I'm thinking of moving the log decoder to the contract instance so that user can decode logs just from receipts that they retrieved via a tx id.

We already have `contaract_logs_abi` and `contract_logs` sway project. He updated both to have a `produce_logs_bad_abi` method. He would just need to add some `asm` in `contract_logs::produce_logs_bad_abi`. Then there would not be any need for `contract_logs_one` and `contract_logs_two`.

Reply:
FYI `LogDecoder` is already inside all contract instances.. we just have to add a getter

https://github.com/FuelLabs/fuels-rs/blob/00e2ada480a2c903b2133cca1c20dadf08c53a5b/packages/fuels-code-gen/src/program_bindings/abigen/bindings/contract.rs#L43

Reply:
@hal3e That makes sense. I thought your suggestion was to use a custom script.